person.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'decorators'
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  private :of_age?
+end
+
+person = Person.new(22, 'maximilianus')
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+puts capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
```

Comment:
- [OPTIONAL] You have a typo on line 33, the variable or method 'capitalizedPerson' does not exist this should be `capitalized_person` instead:
  -  ```capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)```
- Kindly correct this 👍🏻

Reply:
@BenMukebo thank you so much I will correct them soon. Have a nice life.
Kind regards
@ABDULALI3468

Reply:
Thank you!!